client.go

```diff
@@ -15,12 +21,30 @@ const (
 )
 
 type Client struct {
-	cfg *Config
+	cfg    *Config
+	logger *logrus.Logger
 }
 
 func NewClient(c *Config) *Client {
+	var logLevel logrus.Level
+	if c.LogLevel == logLevel {
```

Comment:
if c.debug {
		log.SetLevel(log.DebugLevel)
	} else {
		log.SetLevel(log. ErrorLevel)
	}

Reply:
I would suggest moving this to `NewConfiguration` and read debug level form env. variable.

Reply:
```
// Will log anything that is info or above (warn, error, fatal, panic). Default.
log.SetLevel(log.InfoLevel)
```